### DISCUSSION TITLE
Contribution Process

### DISCUSSION THREAD

**Discussion by CBatemanOracle** [dbeaver/dbeaver#13445] [DBeaver Development]
Hi. Is there a specific process for code contribution, i.e. CLA's etc? 

Thanks,

Cameron

**Comment by luanmateuz** [comment]
They usually accept many external pull requests. You can do one. There are no strict rules for PRs, just follow the existing code style.
If you want to add a new product extension, you can check the plugin https://github.com/dbeaver/dbeaver/tree/devel/test/org.jkiss.dbeaver.ext.test, it contains several basic menu examples / Extensions toolbar.

There are several articles about the development process on wiki (https://github.com/dbeaver/dbeaver/wiki), see the Development section.
